src/domain/thirdpartyRequests/transactions.ts

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * Forwards POST transactions requests to destination FSP for processing
+ * @param {string} path Callback endpoint path
+ * @param {object} headers Headers object of the request 
+ * @param {string} method The http method POST
+ * @param {object} params Params object of the request 
+ * @param {object} payload Body of the POST request
+ * @param {object} span request span
+ * @throws {FSPIOPError} Will throw an error if no endpoint to forward the transactions requests is
+ * found, if there are network errors or if there is a bad response
+ * @returns {Promise<void>}
+ */
+async function forwardTransactionRequest(path: string, headers: Hapi.Util.Dictionary<string>, method: string, params: Hapi.Util.Dictionary<string>, payload?: types.ThirdPartyTransactionRequest, span?: any): Promise<void> {
+
+  const childSpan = span ? span.getChild('forwardTransactionRequest') : undefined
+  const fspiopSource: string = headers[Enum.Http.Headers.FSPIOP.SOURCE]
+  const fspiopDest: string = headers[Enum.Http.Headers.FSPIOP.DESTINATION]
+  const payloadLocal = payload || { transactionRequestId: params.ID }
+  const transactionRequestId: string = (payload && payload.transactionRequestId) || params.ID
+  const endpointType = Enum.EndPoints.FspEndpointTypes.THIRDPARTY_CALLBACK_URL_TRX_REQ_POST
+  let fspiopError: typeof Factory.FSPIOPError | undefined
+  let endpoint: string | undefined
+
+  try {
+    endpoint = await Util.Endpoints.getEndpoint(
+      Config.SWITCH_ENDPOINT,
+      fspiopDest,
+      endpointType)
+    Logger.info(`Resolved PAYER party ${endpointType} endpoint for transactionRequest
+     ${transactionRequestId} to: ${inspect(endpoint)}`)
+    const fullUrl: string = Mustache.render(endpoint + path, { ID: transactionRequestId })
+    Logger.info(`Forwarding transaction request to endpoint: ${fullUrl}`)
+
+    await Util.Request.sendRequest(
+      fullUrl,
+      headers,
+      fspiopSource,
+      fspiopDest,
+      method,
+      method.toUpperCase() !== Enum.Http.RestMethods.GET ? payloadLocal : undefined,
+      Enum.Http.ResponseTypes.JSON,
+      childSpan)
+
+    Logger.info(`Forwarded transaction request ${transactionRequestId} from ${fspiopSource} to ${fspiopDest}`)
+
+    if (childSpan && !childSpan.isFinished) {
+      childSpan.finish()
+    }
+  } catch (err) {
+    Logger.info(`Error forwarding transaction request to endpoint ${endpoint}: ${inspect(err)}`)
+    const errorHeaders = {
+      ...headers,
+      'fspiop-source': Enum.Http.Headers.FSPIOP.SWITCH.value,
+      'fspiop-destination': fspiopSource
+    }
+    fspiopError = ReformatFSPIOPError(err)
+    await forwardTransactionRequestError(
+      errorHeaders,
+      Enum.EndPoints.FspEndpointTemplates.THIRDPARTY_TRANSACTION_REQUEST_PUT_ERROR,
+      Enum.Http.RestMethods.PUT,
+      transactionRequestId,
+      fspiopError!.toApiErrorObject(Config.ERROR_HANDLING.includeCauseExtension, Config.ERROR_HANDLING.truncateExtensions),
```

Comment:
is there a need for a `null assertion operator`? `toApiErrorObject` is already declared in FSPIOPError class
```suggestion
      fspiopError.toApiErrorObject(Config.ERROR_HANDLING.includeCauseExtension, Config.ERROR_HANDLING.truncateExtensions),
```